use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,491 @@
+chipset by executing the following.
+
+### Identify which PCI Bus the Intel QAT card is on
+
+```
+$ for i in 0434 0435 37c8 1f18 1f19; do lspci -d 8086:$i; done
+b1:00.0 Co-processor: Intel Corporation C62x Chipset QuickAssist Technology (rev 04)
+	b3:00.0 Co-processor: Intel Corporation C62x Chipset QuickAssist Technology (rev 04)
+	b5:00.0 Co-processor: Intel Corporation C62x Chipset QuickAssist Technology (rev 04)
+```
+
+### Install necessary bundles for Clear Linux and disable kernel updates
+
+The Clear Linux installer generally installs the `os-core`, `os-core-update`, 
+and `openssh-server` bundle. Add the following minimum bundles including the 
+Linux kernel headers to complete the rest of this guide. The Linux kernel 
+headers are needed to compile the out of tree kernel modules to support Intel 
+QAT on the host. The `os-clr-on-clr` is a large development bundle that can be 
+installed instead to include everything below with additional items. This 
+guide installs the `4.19` LTS kernel from 2018 because the crypto modules are 
+set to `m` and not `y` which allows them to be replaced with the out of tree 
+driver. Otherwise you will need to build your own Clear Linux kernel with the 
+QAT drivers compiled in.
+
+```
+$ sudo swupd autoupdate --disable
+$ sudo swupd bundle-add network-basic kernel-lts2018 linux-lts-dev linux-dev make c-basic go-basic containers-virt dev-utils devpkg-elfutils devpkg-systemd
+$ sudo systemctl start docker
+$ sudo systemctl enable docker
+$ sudo -E gpasswd -a $USER docker
+```
+
+### Replace Clear Linux kernel with LTS kernel
+
+```
+$ lts_kernel=$(sudo clr-boot-manager list-kernels | grep lts2018 | head -n 1)
+$ sudo clr-boot-manager set-kernel $lts_kernel
+$ sudo reboot
+```
+
+### Allow insecure modules to load to host
+
+```
+$ sudo mkdir -p /etc/kernel/cmdline.d
+$ sudo cat << EOF | sudo tee /etc/kernel/cmdline.d/load-modules.conf
+$ module.sig_unenforce
+$ EOF
+$ sudo clr-boot-manager update
+$ sudo reboot
+```
+
+### Download Intel QAT drivers
+
+This will download the Intel QAT drivers from [01.org](https://01.org/intel-quickassist-technology). 
+Make sure to check the website for the latest version
+
+```
+$ mkdir -p $QAT_SRC
+$ cd $QAT_SRC
+$ curl -O $QAT_DRIVER_URL
+$ QAT_DRIVER_NAME=$(echo $QAT_DRIVER_URL | rev | cut -d"/" -f1 | rev)
+$ tar xzvf $QAT_DRIVER_NAME
+```
+
+### Compile Intel QAT drivers against host kernel 
+
+The Linux kernel headers bundle `linux-dev` should already have been added. 
+This replaces any existing modules provided by Clear Linux. If the Clear Linux
+kernel is updated, these modules must be rebuilt against that kernel and 
+reinstalled. By doing it this way, you no longer have to rely on your 
+distribution including the Intel QAT firmware, won’t have to remove the 
+distribution loaded modules each reboot, and get the qat_service helper 
+script. 
+
+```
+$ cd $QAT_SRC
+$ ./configure --disable-qat-lkcf  
+$ sudo -E make all -j$(nproc)
```

Comment:
probably don't need `sudo` on the `make`? And that would maybe annoyingly make files in the tree that are root owned, and a pain to delete later ;-)

Reply:
I verified that I can get rid of sudo -E make all and just have sudo -E make install which is 1 step instead of 2. I have to use sudo in this case because I am installing drivers to parts of the system a user account does not have access to.  I'll check if sudo is needed on the other make where I build against the kata kernel.